File: samples/Test/Classes/Control/PageControlTest.cpp
//
//  PageControlTest.cpp
//  Test
//
//  Created by ly on 2017/3/2.
//
//

#include "PageControlTest.h"

PageControlTest::PageControlTest()
{
    CADrawerController* drawer = (CADrawerController*)CAApplication::getApplication()->getRootWindow()->getRootViewController();
    drawer->setTouchMoved(false);
}

PageControlTest::~PageControlTest()
{
    CADrawerController* drawer = (CADrawerController*)CAApplication::getApplication()->getRootWindow()->getRootViewController();
    drawer->setTouchMoved(true);
}

void PageControlTest::viewDidLoad()
{
    this->getView()->setColor(CAColor_gray);
    
    CAPageControl* pageControl = CAPageControl::createWithLayout(DLayout(DHorizontalLayout_L_R(100, 100), DVerticalLayout_H_C(54, 0.25f)));
    pageControl->setNumberOfPages(5);
    pageControl->setPageIndicatorTintColor(CAColor_white);
    pageControl->setCurrentPageIndicatorTintColor(CAColor_green);
    pageControl->setTarget([=](int index)
    {
        CCLog("btn_tag===%d",pageControl->getCurrentPage());
    });
    this->getView()->addSubview(pageControl);
}

void PageControlTest::viewDidUnload()
{
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

File: samples/Test_JS/Resources/js/Control/PageControlTest.js
/**
 * Created by liangyuan on 2017/3/3.
 */

var PageControlTest = ca.CAViewController.extend({
    ctor: function ()
    {
        this._super();
        this.getView().setColor(ca.color._getGray());

        var pageControl = ca.CAPageControl.createWithLayout(DLayout(DHorizontalLayout_W_C(150, 0.5), DVerticalLayout_H_C(54, 0.25)));
        pageControl.setNumberOfPages(5);
        pageControl.setStyle(ca.CAPageControl.Style.Dot);
        pageControl.setPageIndicatorTintColor(ca.color._getWhite());
        pageControl.setCurrentPageIndicatorTintColor(ca.color._getGreen());
        pageControl.setTarget(function ( func)
        {
            log("btn_tag==="+pageControl.getCurrentPage());
        });
        this.getView().addSubview(pageControl);

        var pageControl2 = ca.CAPageControl.createWithLayout(DLayout(DHorizontalLayout_W_C(150, 0.5), DVerticalLayout_H_C(54, 0.4)));
        pageControl2.setNumberOfPages(5);
        pageControl2.setStyle(ca.CAPageControl.Style.Round);
        pageControl2.setPageIndicatorTintColor(ca.color._getWhite());
        pageControl2.setCurrentPageIndicatorTintColor(ca.color._getGreen());
        pageControl2.setTarget(function ( func)
        {
            log("btn_tag==="+pageControl2.getCurrentPage());
        });
        this.getView().addSubview(pageControl2);

        var pageControl3 = ca.CAPageControl.createWithLayout(DLayout(DHorizontalLayout_W_C(150, 0.5), DVerticalLayout_H_C(54, 0.55)));
        pageControl3.setNumberOfPages(5);
        pageControl3.setStyle(ca.CAPageControl.Style.Rectangle);
        pageControl3.setPageIndicatorTintColor(ca.color._getWhite());
        pageControl3.setCurrentPageIndicatorTintColor(ca.color._getGreen());
        pageControl3.setTarget(function ( func)
        {
            log("btn_tag==="+pageControl3.getCurrentPage());
        });
        this.getView().addSubview(pageControl3);
    },
});